packages/ai/src/ai/web/endpoints/webhook.py

```
@@ -0,0 +1,250 @@
+        logger.info('Executing pipeline %s for token %s', pipeline_id, token)
+        _active_tasks[token]['status'] = 'completed'
+    except Exception:
+        logger.exception('Pipeline execution failed for token %s (pipeline %s)', token, pipeline_id)
+        if token in _active_tasks:
+            _active_tasks[token]['status'] = 'failed'
+    finally:
+        async with _running_lock:
+            _running_count -= 1
+
+
+def register_webhook_routes(server: Any) -> None:
```

Ruff (0.15.7)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `server`

(ANN401)